hw/ip/otbn/data/otbn.ld

```diff
@@ -0,0 +1,39 @@
+/* Copyright lowRISC contributors. */
+/* Licensed under the Apache License, Version 2.0, see LICENSE for details. */
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+
+  OTBN has a pure Harvard architecture, with instruction and data
+  memory both starting at address 0.
+
+  We give each 1MiB of space: too much for the eventual chip - let's
+  come back to this once we know how much space we'll have or (harder)
+  auto-generate it from the address map.
+
+  Because we have the Harvard architecture, we need to alias one of
+  the memories so that LMAs are distinct. We alias instruction memory,
+  putting a load-time version of it at 16 MiB.
+
+*/
+MEMORY
+{
+    imem (x)  : ORIGIN = 0, LENGTH = 1M
+    dmem (rw) : ORIGIN = 0, LENGTH = 1M
+
+    /* Somewhere to put LMAs for data that should end up in IMEM */
+    imem_load (rw) : ORIGIN = 16M, LENGTH = 1M
```

Comment:
It doesn't make much of a difference, but it would be nice to align this layout with the windows into the device we have currently (https://docs.opentitan.org/hw/ip/otbn/doc/index.html#Reg_imem).

This would put imem first with 1MB address space, and then 1MB of dmem right after it.

Theoretically, you could then use the addresses from the ELF, offset them, and load them into the device.

Reply:
That's perfect! And is what I was trying to do - I just hadn't seen that we had a scheme in place. I'll update the patch now.